### PR TITLE
Make deploy runnable on arm64.

### LINKED ISSUE #3107: Make deploy.sh runnable on darwin_arm64
### Context
It's currently unable to run 
`./deploy.sh -u ark -e dev -r us-west-2 -v v1.0.0 -d my-dockerhub-repo` command successfully due to (at least) outdated terraform dependency

### Problem or idea
```
No ccdl-ubuntu-18.04 AMI found for this region, creating a new one
Created new AMI for us-west-2
    name: ccdl-ubuntu-18.04-2022-09-30T09.31.42
    id:   ami-02869d376a418c28c
Initializing modules...
- batch in batch

Initializing the backend...

Successfully configured the backend "s3"! Terraform will automatically
use this backend unless the backend configuration changes.

Initializing provider plugins...
- terraform.io/builtin/terraform is built in to Terraform
- Finding latest version of hashicorp/aws...
- Finding latest version of hashicorp/local...
- Finding latest version of hashicorp/template...
- Installing hashicorp/aws v4.33.0...
- Installed hashicorp/aws v4.33.0 (signed by HashiCorp)
- Installing hashicorp/local v2.2.3...
- Installed hashicorp/local v2.2.3 (signed by HashiCorp)
╷
│ Error: Incompatible provider version
│
│ Provider registry.terraform.io/hashicorp/template v2.2.0 does not have a package available for your current platform, darwin_arm64.
│
│ Provider releases are separate from Terraform CLI releases, so not all providers are available for all platforms. Other versions of this provider may have different platforms supported.
```

### Solution or next step
Update terraform config so the deploy command could be executed successfully.

### PR DESCRIPTION
**Pull request by arkid15r**
## Issue Number

Resolves #3107 

## Purpose/Implementation Notes
Update terraform config:
  - Migrate `template_file` to `templatefile`
  - Inline user_data templates.
  - Fix s3 backend configuration.
  - Set postgres version to 11.16.
  - Change DB `name` to `db_name`.

### CODE REVIEW

File: infrastructure/foreman-configuration/foreman-server-instance-user-data.tpl.sh [hunk 1]
```diff
@@ -77,8 +77,8 @@ docker run \\
        -e DATABASE_PASSWORD=${database_password} \\
        -v /tmp:/tmp \\
        -it ${dockerhub_repo}/dr_\"\$1\" python3 manage.py \"\$2\"
-" >> /home/ubuntu/run_cron_job_test.sh
-chmod +x /home/ubuntu/run_cron_job_test.sh
+" >> /home/ubuntu/run_cron_job.sh
+chmod +x /home/ubuntu/run_cron_job.sh
```
Comment (davidsmejia) [inline infrastructure/foreman-configuration/foreman-server-instance-user-data.tpl.sh:81]: call it run_container_manage_command or something similar, you can pass in the non-dr-prefixed container name and then the name of the mangement command ex: `run_container_manage_command [container] [command]`

If you want to change it I just think you should call it what it does, not named after who is calling it.

File: infrastructure/foreman-configuration/foreman-server-instance-user-data.tpl.sh [hunk 2]
```diff
@@ -112,8 +112,9 @@ service monit restart
 # Install the cron job tests
 crontab -l > tempcron
 cat <<EOF >> tempcron
-0 12 * * MON /bin/bash /home/ubuntu/run_cron_job_test.sh affymetrix check_brainarray_gene_agreement >> /var/log/cron_job_tests.log 2>&1
-0 12 * * MON /bin/bash /home/ubuntu/run_cron_job_test.sh affymetrix check_tx_index_transcript_agreement >> /var/log/cron_job_tests.log 2>&1
+0 12 * * MON /bin/bash /home/ubuntu/run_cron_job.sh affymetrix check_brainarray_gene_agreement >> /var/log/cron_job_tests.log 2>&1
```
Comment (davidsmejia) [inline infrastructure/foreman-configuration/foreman-server-instance-user-data.tpl.sh:115]: apply any name change here as well

**Review by davidsmejia — APPROVED** [review]
This needs to be tested to make sure it works with the version of terraform we use in our deploy scripts.

File: infrastructure/foreman-configuration/foreman-server-instance-user-data.tpl.sh (hunk 2 above)
Comment (arkid15r) [reply infrastructure/foreman-configuration/foreman-server-instance-user-data.tpl.sh]: Done.

File: infrastructure/foreman-configuration/foreman-server-instance-user-data.tpl.sh (hunk 1 above)
Comment (arkid15r) [reply infrastructure/foreman-configuration/foreman-server-instance-user-data.tpl.sh]: Done.

**Pull request merged**